app/models/iiif_canvas_presenter.rb

```diff
@@ -128,6 +128,8 @@ def structure_to_iiif_range
     end
 
     def div_to_iiif_range(div_node)
+      return simple_iiif_range(div_node[:label]) unless div_node.children.present?
```

Comment:
I think this might cause problems if the childless div occurs in an item that has valid spans.  How about narrowing the case where this fix is applied to only be items where there aren't any spans?  I'm not quite sure how to identify that case and still preserve all the childless divs if there is a whole hierarchy of them.  I think it should be possible but might be a tricky or require some refactoring to these recursive methods.

Reply:
I'm not sure it'll be an issue? If there are sibling Divs or Spans to the childless div, they will be in the array of Item children. So an Item with node array `["Childless Div", "Span", "Span"]` should loop through everything and result in `return simple_iiif_range`, `return span_to_iiif_range`, `return span_to_iiif_range`. The return on the childless div shouldn't break the parent Item loop. I've added a test to check the behavior and make sure future changes don't break it, but I think this implementation is safe unless there is something I am missing with `return` and recursive methods.

Reply:
I think the issue won't be on generating the manifest but on the player side consuming the manifest.  From my reading of the IIIF presentation api I think canvas ranges will be played in order so all childless divs replaced with the full canvas range will cause the player play the whole canvas at that point in the structure.  I tried out a hand edited manifest in ramp which had a childless div inserted before the first canvas range.  It played through the whole canvas and then hit an error (about transcripts) when it tried to transition to the next range.  I think Dananji mentioned that Ramp wasn't setup yet to handle this kind of structure. Maybe we could do a screen share to talk through this?